externs/browser/w3c_audio.js

```diff
@@ -292,56 +352,63 @@ AudioDestinationNode.prototype.maxChannelCount;
  */
 function AudioParam() {}
 
-/** @type {number} */
-AudioParam.prototype.value;
+/**
+ * @type {number}
+ * @const
+ */
+AudioParam.prototype.defaultValue;
 
 /**
- * To be deprecated.
+ * @deprecated
  * @type {number}
  */
 AudioParam.prototype.maxValue;
 
 /**
- * To be deprecated.
+ * @deprecated
  * @type {number}
  */
 AudioParam.prototype.minValue;
 
-/** @type {number} */
-AudioParam.prototype.defaultValue;
-
 /**
- * To be deprecated.
+ * @deprecated
  * @type {number}
  */
 AudioParam.prototype.units;
 
+/** @type {number} */
+AudioParam.prototype.value;
+
 /**
- * @param {number} value
  * @param {number} startTime
- * @return {AudioParam}
+ * @return {!AudioParam}
+ * @throws {TypeError} if startTime is negative or not a finite number
  */
-AudioParam.prototype.setValueAtTime = function(value, startTime) {};
+AudioParam.prototype.cancelScheduledValues = function(startTime) {};
 
 /**
  * @param {number} value
  * @param {number} endTime
- * @return {AudioParam}
+ * @return {!AudioParam}
+ * @throws {TypeError} if endTime is negative or not a finite number
  */
-AudioParam.prototype.linearRampToValueAtTime = function(value, endTime) {};
+AudioParam.prototype.exponentialRampToValueAtTime = function(value, endTime) {};
 
 /**
  * @param {number} value
  * @param {number} endTime
- * @return {AudioParam}
+ * @return {!AudioParam}
+ * @throws {TypeError} if endTime is negative or not a finite number
```

Comment:
To avoid ambiguity our internal style guide requires that type annotations always specify explicitly whether a non-primitive type is nullable or not. (Primitive types are non-nullable by default.)

Should this be `{!TypeError}`?